shared_model/backend/protobuf/impl/proto_query_response_factory.cpp

```diff
@@ -52,17 +52,20 @@ namespace {
 
 std::unique_ptr<shared_model::interface::QueryResponse>
 shared_model::proto::ProtoQueryResponseFactory::createAccountAssetResponse(
-    std::vector<std::unique_ptr<shared_model::interface::AccountAsset>> assets,
+    std::vector<std::tuple<interface::types::AccountIdType,
+                           interface::types::AssetIdType,
+                           shared_model::interface::Amount>> assets,
     const crypto::Hash &query_hash) const {
   return createQueryResponse(
       [assets = std::move(assets)](
           iroha::protocol::QueryResponse &protocol_query_response) {
         iroha::protocol::AccountAssetResponse *protocol_specific_response =
             protocol_query_response.mutable_account_assets_response();
-        for (const auto &asset : assets) {
-          *protocol_specific_response->add_account_assets() =
-              static_cast<shared_model::proto::AccountAsset *>(asset.get())
-                  ->getTransport();
+        for (size_t i = 0; i < assets.size(); i++) {
+          auto *asset = protocol_specific_response->add_account_assets();
+          asset->set_account_id(std::move(std::get<0>(assets.at(i))));
```

Comment:
Won't it be better from performance point of view, if you `auto asset = std::move(assets.at(i));`? I think, three `at(i)`-s are an overhead